From: Neerav Parikh <Neerav.Parikh@intel.com>
Date: Tue, 1 Apr 2014 07:11:56 +0000
Subject: [PATCH 058/144] i40e/i40evf: Retrieve and store missing link config
 information
Patch-mainline: v3.16-rc1
Git-commit: 6bb3f23c55bc8c14e9721c64bf51a8290ee50779 (partial)
References: bsc#909486 FATE#317393

Some information returned via "Get Link Status" command was not
being cached in the struct i40e_link_status.
Add this so the driver can utilize this information as needed.

Change-ID: If084a0ae2a63b16b97572196b993742af2c67772
Signed-off-by: Neerav Parikh <neerav.parikh@intel.com>
Acked-by: Greg Rose <gregory.v.rose@intel.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/intel/i40evf/i40e_type.h |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/net/ethernet/intel/i40evf/i40e_type.h
+++ b/drivers/net/ethernet/intel/i40evf/i40e_type.h
@@ -173,6 +173,9 @@ struct i40e_link_status {
 	u8 loopback;
 	/* is Link Status Event notification to SW enabled */
 	bool lse_enable;
+	u16 max_frame_size;
+	bool crc_enable;
+	u8 pacing;
 };
 
 struct i40e_phy_info {
